### DISCUSSION TITLE
specter in bb. vote here for a pod

### DISCUSSION THREAD

**Discussion by stevebuik** [babashka/babashka#913] [Ideas]
I would like to use specter in bb. 

if you would too, chime in here

**Comment by borkdude** [comment]
Perhaps the [specter CLI](https://github.com/borkdude/specter-cli) could expose itself as a pod. 
One challenge is that pods communicate via serialized data (EDN, JSON, or Transit) and the arguments to the Specter API are usually functions. E.g. `(transform [MAP-VALS MAP-VALS] inc {:a {:aa 1} :b {:ba -1 :bb 2}})`. One solution could be that in a pod interface these things could be quoted and then sent as data and evaluated on the pod side.

E.g.
```
(pod/transform '[MAP-VALS MAP-VALS] 'inc {:a {:aa 1} :b {:ba -1 :bb 2}})
```

This might be a bit awkward though.

**Reply by stevebuik** [reply]
Interesting. Could a macro on the bb side do this quoting for us? My macro foo is not strong so I need advice on this :)

I'd be willing to deal with quoting paths and fns to have access to it. 

Would this solution support specter use with thread-last? I use it a lot that way for composing transforms
